shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -306,15 +306,16 @@ private void applyStateToSelection(TextInputChannel.TextEditState state) {
     }
 
     @VisibleForTesting void setTextInputEditingState(View view, TextInputChannel.TextEditState state) {
-        if (!restartAlwaysRequired && !mRestartInputPending && state.text.equals(mEditable.toString())) {
-            applyStateToSelection(state);
+        if (!state.text.equals(mEditable.toString())) {
```

Comment:
So just to make sure I understand, was the previous bug that `mEditable.text` would only be updated when a restart was also required and that it should actually be updated here every time?

This seems to be the new flow, correct me if I'm wrong. I think this could probably use some implementation comments at this point:

1. Replace the entirety of `mEditable.text` with `state` if they differ.
2. Always `applyStateToSelection` to move the cursor. (Is it okay to call this if everything is the same?)
3a. If we don't need to restart, tell `InputMethodManager` about the cursor updates in `mView` via `updateSelection`.
3b. If we DO need to restart, just completely restart the `InputMethodManager` on `view` instead.

Reply:
Yeah, that should be right! I'll document this.